### PR TITLE
Implementation du responsive HomePage

### PR DESCRIPTION
**Pull request by farid1111**
Components:
- [x] - NavBar
- [x] - BlockHeader
- [x] - BlockHictoricDeclarations
- [x] - BlockActualities

Templates:
- [x] - HomepageAgentMobile
- [x] - HomepageAgentDesktop

Page:
- [x] - agent/index.tsx

Utils:
- [x] - useResponsive

## Summary by CodeRabbit

- **New Features**
	- Introduced `BlockActualities` component for listing news and updates with responsive design adjustments.
	- Added `BlockHeaderAgent` component with search functionality and information display.
	- Launched `BlockHistoricDeclarations` component to show historical declarations with categorization and responsive layout.
	- Implemented `NavBar` component for improved navigation experience.
	- Expanded `SvgIcon` component with new icon support.
	- Enhanced `TitleAgent` and `Typography` components with size and text wrapping customization options.

- **Enhancements**
	- Improved responsiveness and layout adjustments across various components for better user experience on different screen sizes.
	- Added Storybook stories for new components to facilitate testing and showcase.

- **Bug Fixes**
	- Fixed search functionality to properly initialize based on query parameters.

- **Refactor**
	- Streamlined agent homepage rendering logic based on screen size without separate scanner and tools modes.

- **Style**
	- Updated `LinkWithIcon` component design with optional border and rounded shape.
	- Added new color definition for consistent navbar styling.

- **Documentation**
	- Exported new constants and enums for better code management and reuse.

### CODE REVIEW

File: front/src/components/common/BlockActualities/BlockActualities.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { ActualityCard } from '@/components/business/ActualityCard';
+
+export interface BlockActualitiesProps {
+  isLoading?: boolean;
+  data?: ActualityResponse[];
+}
+
+export const BlockActualities: React.FC<BlockActualitiesProps> = ({
+  isLoading,
+  data,
```
Comment (sabalot) [inline front/src/components/common/BlockActualities/BlockActualities.tsx]: Renommer en actualities

File: front/src/components/common/BlockHistoricDeclarations/BlockHistoricDeclarations.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  paidDeclarations,
+  isErrorDeclarationsLoading,
+  errorDeclarations,
+}: BlockHistoricDeclarationsProps) => {
+  const router = useRouter();
+  return (
+    <div className="flex flex-col justify-start gap-[30px]">
+      <TitleAgent title="Historique des déclarations" textPosition="text-left" size="text-3xl" />
+      <div className="flex flex-row items-center gap-10">
+        {!isSubmittedDeclarationsLoading && submittedDeclarations && (
```
Comment (sabalot) [inline front/src/components/common/BlockHistoricDeclarations/BlockHistoricDeclarations.tsx]: Ajouter le skeleton + laisser toujours le titre afficher

File: front/src/components/common/Typography/style/typography.style.ts [hunk 3]
```diff
@@ -144,6 +144,13 @@ export const getTruncate = (truncate?: boolean): string => {
   return '';
 };
 
+export const getNoWrap = (noWrap?: boolean): string => {
+  if (noWrap) {
+    return 'whitespace-nowrap';
+  }
+  return '';
+};
+
```
Comment (sabalot) [inline front/src/components/common/Typography/style/typography.style.ts:153]: tu peux retirer

**Comment by sabalot** [comment]
@CodiumAI-Agent /review

**Comment by sabalot** [comment]
@CodiumAI-Agent /improve

File: front/src/pages/agent/index.tsx [hunk 4]
```diff
@@ -1,238 +1,19 @@
...
-    setScanningStarted(false);
-  };
-
-  const getDeclarationMutation = useDeclarationMutation({
-    onSuccess: ({ data }) => {
-      router.push(`/agent/declaration/${data.id}`);
-      stopCamera();
-    },
+  const isDesktopOrLaptop = useMediaQuery({
+    query: '(min-width: 1121px)',
```
Comment (sabalot) [inline front/src/pages/agent/index.tsx]: utiliser une const

**Review by sabalot — CHANGES_REQUESTED** [review]

File: front/src/components/common/Typography/style/typography.style.ts (hunk 3 above)
Comment (farid1111) [reply front/src/components/common/Typography/style/typography.style.ts:153]: j'en ai besoin en fait

**Review by Touan95 — APPROVED** [review]

**Pull request merged**